core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -225,9 +301,16 @@ public void accept(Object o) throws Throwable {
             Object key = marshaller.objectFromByteBuffer(serializedKey);
             int segment = keyPartitioner.getSegment(key);
 
-            int indexedOffset = header.valueLength() > 0 ? scheduledOffset : ~scheduledOffset;
-            boolean drop = true;
+            int valueLength = header.valueLength();
+            int indexedOffset = valueLength > 0 ? scheduledOffset : ~scheduledOffset;
+            // Whether to drop the entire index (this cannot be true if truncate is false)
+            // We drop all entries by default unless it is a log file as we can't drop any of those since we may
+            // try to compact a log file multiple times, note modifications to drop variable below should only be to set
+            // it to false
+            boolean drop = !isLogFile;
+            // Whether to truncate the value
             boolean truncate = false;
+            // Whether the entry when truncating was due to expiration (this cannot be true if truncate is false)
```

Comment:
Outdated comment? It doesn't seem relevant to `entry`.

Reply:
Removed